opensearchpy/connection/http_async.py

```diff
@@ -190,17 +193,23 @@ async def perform_request(
             body = self._gzip_compress(body)
             req_headers["content-encoding"] = "gzip"
 
-        req_headers = {
-            **req_headers,
-            **self._http_auth(method, url, query_string, body),
-        }
+        auth = None
+        if self._http_auth:
+            if isinstance(self._http_auth, tuple):
```

Comment:
This shouldn't be possible anymore because `__init__` turned everything into `aiohttp.BasicAuth`. Is that callable, too?

Reply:
It _can_ be a callable if it's an instance of the AWS auth class, which doesn't belong in the `auth` parameter to the session, so it's covering that base.

This would simplify down to

```
auth = self._http_auth if isinstance(self._http_auth, aiohttp.BasicAuth) else None
```

We still need to maintain `auth=None` to the session if the AWS auth class was passed in.